packages/cli/generators/service/index.js

```diff
@@ -0,0 +1,242 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/cli
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+const _ = require('lodash');
+const ArtifactGenerator = require('../../lib/artifact-generator');
+const fs = require('fs');
+const debug = require('../../lib/debug')('service-generator');
+const inspect = require('util').inspect;
+const path = require('path');
+const chalk = require('chalk');
+const utils = require('../../lib/utils');
+
+const VALID_CONNECTORS_FOR_SERVICE = ['Model'];
+
+const SERVICE_TEMPLATE = 'service-template.ts.ejs';
+
+const PROMPT_MESSAGE_DATA_SOURCE = 'Please select the datasource';
+
+const ERROR_NO_DATA_SOURCES_FOUND = 'No datasources found at';
+
+module.exports = class ServiceGenerator extends ArtifactGenerator {
+  // Note: arguments and options should be defined in the constructor.
+  constructor(args, opts) {
+    super(args, opts);
+  }
+
+  _setupGenerator() {
+    this.artifactInfo = {
+      type: 'service',
+      rootDir: utils.sourceRootDir,
+    };
+
+    this.artifactInfo.outDir = path.resolve(
+      this.artifactInfo.rootDir,
+      utils.servicesDir,
+    );
+
+    this.artifactInfo.datasourcesDir = path.resolve(
+      this.artifactInfo.rootDir,
+      utils.datasourcesDir,
+    );
+
+    this.artifactInfo.defaultTemplate = SERVICE_TEMPLATE;
+
+    this.option('datasource', {
+      type: String,
+      required: false,
+      description: 'A valid datasource name',
+    });
+
+    return super._setupGenerator();
+  }
+
+  setOptions() {
+    return super.setOptions();
+  }
+
+  async checkPaths() {
+    // check for datasources
+    if (!fs.existsSync(this.artifactInfo.datasourcesDir)) {
+      return this.exit(
+        new Error(
+          `${ERROR_NO_DATA_SOURCES_FOUND} ${this.artifactInfo.datasourcesDir}.
+        ${chalk.yellow(
```

Comment:
I find this line break rather suspicions, because the indenting of the second line is going to be preserved in the message.

```
No datasources found at src/datasources.
         Please visit https://loopback.io/doc/en/lb4/DataSource-generator.html for information on how datasources are discovered
```

Please find a better way how to compose this error message. For example:

```ts
const message = [
  `${ERROR_NO_DATA_SOURCES_FOUND} ${this.artifactInfo.datasourcesDir}.`,
  chalk.yellow(
    'Please visit https://loopback.io/doc/en/lb4/DataSource-generator.html for information ' + 
      'on how datasources are discovered'
  ),
].join('\n');
return this.exit(new Error(message));
```